.github/workflows/publish-latest.yml

```diff
@@ -87,3 +87,25 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+  log-failure-metric:
+    # Send data point 1 to metric PublishLatestFailure, if failure occurs
+    uses: aws-amplify/amplify-ui/.github/workflows/reusable/metric.yml
+    with:
+      status: ${{ failure() }}
+      metric-name: PublishNPMFailure
+      value: 1
+    secrets:
+      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+
+  log-success-metric:
+    # Send data point 0 to metric PublishLatestFailure, if success
+    uses: aws-amplify/amplify-ui/.github/workflows/reusable/metric.yml
```

Comment:
For example, we can have:

```yml
  log-metric:
    # Send data point 0 or 1 to metric PublicLatestFailure
    uses: aws-amplify/amplify-ui/.github/workflows/reusable/metric.yml
    with:
      is-success: ${{ success() }} # This will be true if publish succeeded, false otherwise.
      metric-name: PublishNPMFailure 
      success-value: 0 # this is the value that will be sent when status is 'true'
      fail-value: 1 # this is the value that will be sent when status is 'false'
    secrets:
      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
```

Reply:
That was my first thought, but the reason I didn't do it was because it would be more flexible to only have 1 type of datapoint logged each time, because we may have more than failures and successes in the future. Potentially, we can other numbers, status, etc.

Reply:
that's a very good point actually! I'm convinced.

Reply:
We can abstract more later as needed, this is a good start.

Reply:
Yea. I though about it. It's good for now and easy to be changed later.